crates/ui/src/windows/event_edit.rs

```diff
@@ -22,55 +22,252 @@
+            switch_2_modal: database_modal::Modal::new(id_source.with("switch_2_modal")),
+            variable_modal: database_modal::Modal::new(id_source.with("variable_modal")),
+            graphic_modal,
         }
     }
 }
 
 impl luminol_core::Window for Window {
     fn name(&self) -> String {
-        format!("Event: {}, {} in Map {}", self.name, self.id, self.map_id)
+        format!("Event '{}' ID {}", self.event.name, self.event.id)
     }
 
     fn id(&self) -> egui::Id {
         egui::Id::new("luminol_event_edit")
             .with(self.map_id)
-            .with(self.id)
+            .with(self.event.id)
     }
 
-    // This needs an overhaul
     fn show(
         &mut self,
-        _ctx: &egui::Context,
-        _open: &mut bool,
-        _update_state: &mut luminol_core::UpdateState<'_>,
+        ctx: &egui::Context,
+        open: &mut bool,
+        update_state: &mut luminol_core::UpdateState<'_>,
     ) {
-        todo!()
+        let mut win_open = true;
+        let mut needs_save = false;
+
+        egui::Window::new(self.name())
+            .open(&mut win_open)
+            .id(self.id())
+            .show(ctx, |ui| {
+                let id_source = self.id();
+                let previous_page = self.selected_page;
+                egui::TopBottomPanel::top(id_source.with("top_panel")).show_inside(ui, |ui| {
+                    ui.horizontal(|ui| {
+                        ui.label("Name: ");
+                        ui.text_edit_singleline(&mut self.event.name);
+                    });
+                    ui.horizontal(|ui| {
+                        ui.label("Page: ");
+                        for i in 0..self.event.pages.len() {
+                            ui.selectable_value(&mut self.selected_page, i, format!("{}", i + 1));
+                        }
+
+                        if ui
+                            .button(egui::RichText::new("Add").color(egui::Color32::LIGHT_GREEN))
+                            .clicked()
+                        {
+                            self.event.pages.push(rpg::EventPage::default());
+                            self.selected_page = self.event.pages.len() - 1;
+                        }
+
+                        let button = egui::Button::new(
+                            egui::RichText::new("Delete").color(egui::Color32::LIGHT_RED),
+                        );
+                        if ui.add_enabled(self.event.pages.len() > 1, button).clicked() {
+                            self.event.pages.remove(self.selected_page);
+                            self.selected_page = self.selected_page.saturating_sub(1);
+                        }
+                        if ui.button(egui::RichText::new("Clear")).clicked() {
+                            self.event.pages[self.selected_page] = rpg::EventPage::default();
+                        }
+                    });
+                });
```

Comment:
Can you add a little bit of spacing above and below this top panel in the event editor window? It looks a bit odd when the bottom edges of the buttons in the top panel are flush with the separator. Same goes for the top panel in the graphic picker, although that one only needs extra spacing above.